src/NServiceBus.Core/SettingsExtensions.cs

```diff
@@ -20,16 +20,19 @@ public static T GetConfigSection<T>(this ReadOnlySettings settings) where T : cl
             var typesToScan = settings.GetAvailableTypes();
             var configurationSource = settings.Get<IConfigurationSource>();
 
-            // ReSharper disable HeapView.SlowDelegateCreation
-            var sectionOverrideType = typesToScan.Where(t => !t.IsAbstract)
-                .FirstOrDefault(t => typeof(IProvideConfiguration<T>).IsAssignableFrom(t));
-            // ReSharper restore HeapView.SlowDelegateCreation
+            var sectionOverrideTypes = typesToScan.Where(t => !t.IsAbstract && typeof(IProvideConfiguration<T>).IsAssignableFrom(t)).ToArray();
+            if (sectionOverrideTypes.Length > 1)
+            {
+                throw new Exception($"Ambiguous configuration providers implementing IProvideConfiguration<{typeof(T).Name}> were found. Ensure that only one configuration provider per configuration section is found to resolve this error.");
```

Comment:
The wording "Ambiguous configuration providers ... were found" seems a bit off to me. We didn't find "ambiguous" implementations, we found more than one implementation, so that makes it ambiguous as to which one we should use.

I suggest changing this to "More than one configuration providers ... were found", we could also possibly include the number of them we found, to help users know how many they need to fix.